cpp/csp/adapters/websocket/ClientConnectionRequestAdapter.cpp

```diff
@@ -0,0 +1,48 @@
+#include <csp/adapters/websocket/ClientConnectionRequestAdapter.h>
+#include <csp/python/Conversions.h>
+#include <Python.h>
+
+namespace csp::adapters::websocket {
+
+ClientConnectionRequestAdapter::ClientConnectionRequestAdapter(
+    Engine * engine,
+    WebsocketEndpointManager * websocketManager,
+    bool is_subscribe,
+    size_t caller_id,
+    boost::asio::strand<boost::asio::io_context::executor_type>& strand
+
+) : OutputAdapter( engine ),  
+    m_websocketManager( websocketManager ),
+    m_strand( strand ),
+    m_isSubscribe( is_subscribe ),
+    m_callerId( caller_id ),
+    m_checkPerformed( is_subscribe ? false : true )  // we only need to check for pruned input adapters
+{}
+
+void ClientConnectionRequestAdapter::executeImpl()
+{
+    // One-time check for pruned status
+    if (unlikely(!m_checkPerformed)) {
+        m_isPruned = m_websocketManager->adapterPruned(m_callerId);
+        m_checkPerformed = true;
+    }
+
+    // Early return if pruned
+    if (unlikely(m_isPruned))
+        return;
+
+    auto raw_val = input()->lastValueTyped<PyObject*>();
+    auto val = python::fromPython<std::vector<Dictionary>>(raw_val);
```

Comment:
there shouldnt be any python related code on your cpp adapters

Reply:
But the input is a python iist of python dictionaries, should I make another csp struct to replace that? (Maybe an 'InternalConnectionRequest' struct?). I'd like to keep the processing in python, which is why i didn't pass the 'ConnectiomRequest' structs directly

Reply:
Ok I switched to just passing a string for headers, and using rapidjson to update the headers in c++. I added a new struct to avoid python.